lux/lib/lux/prisms/discord/messages/edit_message_prism.ex

```diff
@@ -0,0 +1,150 @@
+    description: "Edits a message in a Discord channel",
+    input_schema: %{
+      type: :object,
+      properties: %{
+        channel_id: %{
+          type: :string,
+          description: "The ID of the channel containing the message",
+          pattern: "^[0-9]{17,20}$"
+        },
+        message_id: %{
+          type: :string,
+          description: "The ID of the message to edit",
+          pattern: "^[0-9]{17,20}$"
+        },
+        content: %{
+          type: :string,
+          description: "The new content for the message",
+          minLength: 1,
+          maxLength: 2000
+        }
+      },
+      required: ["channel_id", "message_id", "content"]
+    },
+    output_schema: %{
+      type: :object,
+      properties: %{
+        message: %{
+          type: :object,
+          properties: %{
+            id: %{type: :string, description: "Message ID"},
+            content: %{type: :string, description: "Updated message content"},
+            channel_id: %{type: :string, description: "Channel ID where message was edited"},
+            edited_timestamp: %{type: :string, description: "When the message was edited"},
+            author: %{
+              type: :object,
+              properties: %{
+                id: %{type: :string, description: "Author's Discord ID"},
+                username: %{type: :string, description: "Author's username"}
+              }
+            }
+          },
+          required: ["id", "content", "edited_timestamp"]
+        }
+      },
+      required: ["message"]
+    }
+
+  alias Lux.Lenses.DiscordLens
+  require Logger
+
+  # Discord API error codes
+  @discord_errors %{
+    10003 => "Unknown channel",
+    10008 => "Unknown message",
+    50001 => "Missing access",
+    50005 => "Cannot edit this message",
+    50013 => "Missing permissions",
+    50035 => "Invalid form body",
+    40005 => "Request entity too large",
+    50006 => "Cannot send empty message"
+  }
+
+  @doc """
+  Handles the request to edit a message in a Discord channel.
+  """
+  def handler(%{channel_id: channel_id, message_id: message_id, content: content} = input, %{agent: agent} = _ctx) do
+    case validate(input) do
+      :ok ->
+        Logger.info("Agent #{agent.name} editing message #{message_id} in channel #{channel_id}")
+
+        case DiscordLens.focus(%{
+          endpoint: "/channels/#{channel_id}/messages/#{message_id}",
+          method: :patch,
+          body: %{content: content}
+        }) do
+          {:ok, response} ->
+            Logger.info("Successfully edited message #{message_id} in channel #{channel_id}")
+            {:ok, %{message: response}}
+          {:error, reason} ->
+            Logger.error("Failed to edit Discord message: #{inspect(reason)}")
+            handle_discord_error(reason)
+        end
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  @doc """
+  Validates the input parameters.
+  """
+  def validate(input) do
```

Comment:
We actually have `ExJsonSchema.Validator.validate` to validate the inputs. You'd need to look at [the docs](https://github.com/jonasschmidt/ex_json_schema) to see how schema resolution works (for example used [here](https://github.com/sojinmm/lux/blob/d18fff52c8c968fc414d233fd86dff8d11348c4e/lux/lib/lux/signal_schema.ex#L85-L86) and [here](https://github.com/sojinmm/lux/blob/d18fff52c8c968fc414d233fd86dff8d11348c4e/lux/lib/lux/signal_schema.ex#L139-L140) but it should be better than writing the validation and error handling yourself.